src/address/normalize-address.ts

```diff
@@ -0,0 +1,65 @@
+import { EventEmitter } from "../isomorphic.node";
+import { NormalizedConfig } from "../config";
+import {
+  AddressValidateParams,
+  AddressValidateResult,
+  callJsonRpcMethod,
+} from "../json-rpc";
+import { formatAddress } from "./format-address";
+import {
+  Address,
+  AddressValidationResult,
+  NormalizedAddress,
+} from "./public-types";
+import { validateAddress } from "./validate-address";
+import { validateInputAddress } from "./validate-input-address";
+import { ShipEngineError } from "../errors";
+import { ErrorCode, ErrorType } from "../enums";
+
+/**
+ * Normalizes an address and returns it.
+ */
+export async function normalizeAddress(
+  address: Address,
+  config: NormalizedConfig,
+  events: EventEmitter
+  // @ts-expect-error TypeScript is confused by the return in the for loop
```

Comment:
This comment is irrelevant

Reply:
This is the line I wanted to talk to you about. Because my return statements are inside conditional blocks, TS thinks I'm not returning a value.

Reply:
I can't resolve this one. Can we pair on this?